packages/runed/src/lib/functions/box/box.svelte.ts

```diff
@@ -0,0 +1,194 @@
+import type { Expand, Getter } from "$lib/internal/types.js";
+import { isFunction, isObject } from "$lib/internal/utils/is.js";
+
+const BoxSymbol = Symbol("box");
+const isWritableSymbol = Symbol("is-writable");
+
+export interface ReadableBox<T> {
+	readonly [BoxSymbol]: true;
+	readonly value: T;
+}
+
+export interface WritableBox<T> extends ReadableBox<T> {
+	readonly [isWritableSymbol]: true;
+	value: T;
+}
+
+/**
+ * @returns Whether the value is a Box
+ */
+function isBox(value: unknown): value is ReadableBox<unknown> {
+	return isObject(value) && BoxSymbol in value;
+}
+/**
+ * @returns Whether the value is a WritableBox
+ */
+function isWritableBox(value: unknown): value is WritableBox<unknown> {
+	return isBox(value) && isWritableSymbol in value;
```

Comment:
just to make the usage a bit more consisent with the rest
```suggestion
	return box.isBox(value) && isWritableSymbol in value;
```